cpp/src/arrow/array/data.cc

```diff
@@ -195,6 +197,7 @@ std::shared_ptr<ArrayData> ArrayData::Slice(int64_t off, int64_t len) const {
   } else {
     copy->null_count = null_count != 0 ? kUnknownNullCount : 0;
   }
+  copy->statistics = nullptr;
```

Comment:
A naive question, `if off == 0 && len == this->length`, should we keep this?

Reply:
Hmm. I think that we don't need to do it:

* "Slice() always discards statistics" is simple and easy to understand
* In general, `Slice(0, length)` will not be used because it's just a `Copy()` and an user should use `Copy()` instead of `Slice(0, length)`

If it may be useful, we can revisit this later.

Reply:
Could we add comment in `statistics` or subslice? This would helps user on this?

Reply:
It's a good idea. I've add it.